src/jit/valuenum.cpp

```diff
@@ -7033,13 +7033,17 @@ void Compiler::fgValueNumberTree(GenTree* tree, bool evalAsgLhsInd)
                 break;
 
                 case GT_NULLCHECK:
+                {
                     // Explicit null check.
-                    tree->gtVNPair =
-                        vnStore->VNPWithExc(ValueNumPair(ValueNumStore::VNForVoid(), ValueNumStore::VNForVoid()),
-                                            vnStore->VNPExcSetSingleton(
-                                                vnStore->VNPairForFunc(TYP_REF, VNF_NullPtrExc,
-                                                                       tree->gtOp.gtOp1->gtVNPair)));
-                    break;
+                    // Handle case where operand tree also may case exceptions.
+                    ValueNumPair excSet = vnStore->VNPExcSetSingleton(
+                        vnStore->VNPairForFunc(TYP_REF, VNF_NullPtrExc,
+                                               vnStore->VNPNormVal(tree->gtOp.gtOp1->gtVNPair)));
+
+                    excSet         = vnStore->VNPExcSetUnion(excSet, vnStore->VNPExcVal(tree->gtOp.gtOp1->gtVNPair));
```

Comment:
I think that we should declare a new local excSetBoth or similar instead of stomping on the old value of excSet

Reply:
Ok, will fix.